src/go/rdctl/pkg/factoryreset/factory_reset_windows.go

```diff
@@ -98,6 +80,53 @@ func getDirectoriesToDelete(keepSystemImages bool) ([]string, error) {
 	return dirs, nil
 }
 
+func GetLockfilePath(appName string) (string, error) {
+	appData, err := getKnownFolder(windows.FOLDERID_RoamingAppData)
+	if err != nil {
+		return "", fmt.Errorf("could not get AppData folder: %w", err)
+	}
+	return path.Join(appData, appName, "lockfile"), nil
+}
+
+func UnregisterWSL() error {
+	cmd := exec.Command("wsl", "--list", "--quiet")
+	cmd.SysProcAttr = &syscall.SysProcAttr{CreationFlags: 0x08000000} // CREATE_NO_WINDOW
+	rawBytes, err := cmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("error getting current WSLs: %w", err)
+	}
+	decoder := unicode.UTF16(unicode.LittleEndian, unicode.IgnoreBOM).NewDecoder()
+	cookedBytes, err := decoder.Bytes(rawBytes)
+	if err != nil {
+		return fmt.Errorf("error decoding ucs2 output: %w", err)
+	}
+	asciiOutput := strings.ReplaceAll(string(cookedBytes), "\r", "")
+	wsls := strings.Split(asciiOutput, "\n")
+	wslsToKill := []string{}
+	for _, s := range wsls {
+		if s == "rancher-desktop" || s == "rancher-desktop-data" {
+			wslsToKill = append(wslsToKill, s)
+		}
+	}
+
+	for _, wsl := range wslsToKill {
+		cmd := exec.Command("wslconfig", "/u", wsl)
+		cmd.SysProcAttr = &syscall.SysProcAttr{CreationFlags: 0x08000000} // CREATE_NO_WINDOW
+		err := cmd.Run()
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Error unregistering WSL %s: %s\n", wsl, err)
+		}
+	}
+	return nil
+}
+
+func PowershellKillWindows() {
+	cmd := exec.Command("powershell", "-WindowStyle", "hidden", "-Command",
+		`Stop-Process -Name "Rancher Desktop" -ErrorAction "SilentlyContinue"`)
+	cmd.SysProcAttr = &syscall.SysProcAttr{CreationFlags: 0x08000000} // CREATE_NO_WINDOW
```

Comment:
We're using `CREATE_NO_WINDOW` quite a few times; it's probably time to put it into a constant.

Reply:
Yup